src/Tasks/Microsoft.NET.Build.Tasks/ProcessFrameworkReferences.cs

```diff
@@ -230,58 +229,74 @@ protected override void ExecuteCore()
                     isTrimmable = selectedRuntimePack?.IsTrimmable;
                 }
 
-                //  Only add runtime packs where the framework reference name matches the RuntimeFrameworkName
-                //  Framework references for "profiles" will use the runtime pack from the corresponding non-profile framework
+                bool includeInPackageDownload;
+                KnownRuntimePack runtimePackForRuntimeIDProcessing;
                 if (knownFrameworkReference.Name.Equals(knownFrameworkReference.RuntimeFrameworkName, StringComparison.OrdinalIgnoreCase))
                 {
-                    bool processedPrimaryRuntimeIdentifier = false;
+                    //  Only add runtime packs where the framework reference name matches the RuntimeFrameworkName
```

Comment:
Ideally these comments should be able to express in code. But meanwhile this class should really split into 4-5 class to make things clear. I will try to refactor that. (you should try too if you have time. reference http://arlobelshee.com/the-core-6-refactorings/)